app/controllers/items/new.js

```diff
@@ -70,6 +71,20 @@ export default GoodcityController.extend(
       return this.get("i18n").t(str);
     },
 
+    allAvailablePrinters: Ember.computed(function() {
+      return this.get("printerService").allAvailablePrinters();
+    }),
+
+    selectedPrinterDisplay: Ember.computed("selectedPrinterId", function() {
+      const printerId = this.get("selectedPrinterId");
+      if (printerId) {
+        const printer = this.store.peekRecord("printer", printerId);
+        return { name: printer.get("name"), id: printer.id };
+      } else {
+        return this.get("allAvailablePrinters")[0];
```

Comment:
A bit odd that `allAvailablePrinters` returns a list of objects `{name, id}`
This object structure is only needed in the page for the dropdown list (I assume) so you can build it here rather than in the service.

Otherwise, one day someone will call your service method, expecting a printer model, but instead will receive this object

Reply:
@patrixr i made the changes there is not need of creating a object i can use peekAll printer as well.